src/service.c

```diff
@@ -70,6 +71,11 @@ static gboolean r_on_handle_install_bundle(
 	args->notify = service_install_notify;
 	args->cleanup = service_install_cleanup;
 
+	g_variant_dict_init(&dict, arg_args);
+	g_variant_dict_lookup(&dict, "ignore-compatible", "b", &args->ignore_compatible);
+	arg_args = g_variant_dict_end(&dict);
+	g_variant_unref(arg_args);
```

Comment:
This is a bit strange. I'm not sure if `g_variant_dict_init()` consumes the ref for `arg_args` or not, but either way, `g_variant_dict_clear()` should be used here and maybe `g_variant_unref()` if necessary.

Reply:
Also, you might want to change the commit order: Before this change, `arg_args` was leaked.

Reply:
Right, `g_variant_dict_clear()` should do it here then there's no need to unref (and abuse `arg_args`).

About the `arg_args` leakage: This is passed from outside and thus I would expect the caller to free it.

Reply:
That depends on how the API is defined. And the caller is generated gdbus code, right?